### DISCUSSION TITLE
Multi-Scrobbler vs. Panoscrobbler

### DISCUSSION THREAD

**Discussion by mavproductions** [FoxxMD/multi-scrobbler#213] [General]
Some clear differences, but curious what this could accomplish over a lot of apps with built in scrobbling options + panoscrobbler for on-the-go!

Will be trying this out regardless! :)

**Comment by FoxxMD** [comment]
The main difference is that MS is **listening device independent.** It talks **directly to the music service APIs** to track your listening activity rather than needing to live on each device you are using.

Consider this scenario:

* You use Spotify and [Navidrome](https://www.navidrome.org/) (subsonic) for your music
* You listen to Spotify on
   * laptop at work using the web player on Chrome because your work doesn't let you install the native app
   * home PC using the native app
   * android phone
   * android tv
* You listen to subsonic on
  * android phone using Symphonium
  * navidrome on home PC/laptop using web player
  * [Soniclair](https://github.com/thelinkin3000/soniclair) for android tv
* You want to scrobble to
  * last.fm
  * maloja

### Platform/App Scrobbling

With apps + built in scrobbling the listening activity that is captured is specific to the device or app you are using. This applies to Panoscrobbler too -- it uses the Android media API to capture all(?) listening activity **on your specific android device** but that is as far as it can go.

```mermaid
flowchart LR
   A[Spotify]  --> B
   F[Navidrome] --> G
   subgraph Android
    B{Spotify App} --> C[User]
    G{Symphonium} --> C
    C -->|Android Media API| D
	D@{ shape: lin-rect, label: "Pano Scrobbler" } --> E[Last.fm]
    D --> M[Maloja]
   end
```

The same goes for all the other platforms I listed in the scenario so...

* spotify will scrobble to last.fm anywhere but not maloja
  * Install [Web Scrobbler](https://chromewebstore.google.com/detail/web-scrobbler/hhinaapppaileiechjoiifaancjggfjm) chrome extension to cover maloja on laptop but...
    * Need to find a desktop app for spotify native on home PC? will it scrobble to maloja?
  * How to scrobble to maloja when playing on TV?
* Navidrome will scrobble to last.fm when using web player
  * Web Scrobbler will cover maloja for web player as well 
  * What about tv with Soniclair?

Now...what if you want to use a different subsonic server with a nicer web player? Will it scrobble to last.fm? Will Web Scrobbler recognize it? What if you switch to an iphone, will it have as good a scrobbling app? What if you get a new laptop or re-install windows, now you need to re-install extensions/scrobble apps and then sign in to all your scrobble services again.

What about scrobbling back to maloja when you are out of the house? It needs to have connectivity so panoscrobbler can send it scrobbles so it either needs to be on the public internet or you need to set up a vpn/tailscale and connect it to your phone.

This is a huge balancing act to do something (track your listening history) that you really don't want to have to think about. It should be set-and-forget...

### Multi-scrobbler

Multi-scrobbler is not an app that is installed on a device or run on your machine to capture listening from that machine. Instead, it uses the music service's API to directly to talk to your music services and capture your listening history **anywhere you are listening.**

```mermaid
flowchart LR
    A[Spotify] --> d1
    A --> d2
    A --> d3
    %%A[Spotify] --> Android
    %%A --> Laptop
    F[Navidrome] --> d5
    F --> d4
    %%F[Navidrome] --> Android
    %%F --> Laptop
    subgraph Android
	    d1{Spotify App} <--> c1(User)
	    d5{Symphonium} <--> c1
    end
    subgraph Laptop
	    d2{Spotify App} <--> c2(User)
	    d3{Spotify Web} <--> c2
	    d4{Navidrome Web} <--> c2
    end
    A <-->|Spotify API| ms@{ shape: lin-rect, label: "multi-scrobbler" }
    F <-->|Subsonic API| ms
   ms --> l[Last.fm]
   ms --> mal[Maloja]   
```

So MS can scrobble for you regardless of where you are playing your music. There's no configuration or additional apps to install on each device you use. You won't have to think about whether you are on android, iphone, tv, laptop, pc, whatever. And since MS lives in your home network and only talks to music service APIs you don't need it to be accessible to the public internet to work when you are away from your LAN.

#### Scrobble Mirror

The other advantage to MS is that it can **mirror scrobbles from one service to another.**

Imagine you have an app like Deezer that can scrobble to Last.fm but you also want your scrobbles on Maloja for data backup. MS can [monitor Last.fm for new scrobbles](https://foxxmd.github.io/multi-scrobbler/docs/configuration/#lastfm-source) and then mirror those to Maloja so you get a backup.

## Not Competitors

MS and Panoscrobbler/other apps are not mutually exclusive though. While MS is superior for platforms that enable tracking activity -- like Spotify, subsonic, Jellyfin... -- there are other platforms that do not surface this information easily -- like Youtube Music or Deezer -- which are better serviced by app-specific scrobblers. A common scenario for MS users is:

* use MS to directly scrobble for Spotify, Jellyfin/Plex to Maloja/Last.fm
* use Panoscrobbler and Web Scrobbler for scrobbling from Youtube Music to Maloja/Last.fm

so they can go hand-in-hand to make sure your listening habits are fully covered. MS has loads of additional configuration to help stay out of the way for mixed scrobbling as well -- you can specify that some Sources only scrobble to some Clients. Or in most Sources you can filter out listening activity based on user or device.

At the end of the day MS is designed to help you scrobble from everywhere to anywhere without causing a headache. That can entail using multiple technologies alongside MS so its built to be adaptable to that as much as possible.

**Comment by mavproductions** [comment]
I'm having issues with Plexamp right now with my current scrobbling setup. 

Can multi-scrobbler recognize duplicate scrobbles to last.fm currently if there is ever a collision with another scrobbling app?

**Edit**: My problem is I use plexamp from my phone then cast it to my laptop running plexamp. If I want to use MS to manage my laptop plexamp scrobbles, there can be potential panoscrobbler issues, and I really like panoscrobbler for my plexamp on the fly scrobbling. The issue is panoscrobbler obviously can't differentiate my phones (device) speakers and laptop (device) audio output, so it can't know when not to scrobble.

Plex API scrobbling is sometimes delayish and ideally on discord, I would like to have my live last.fm scrobbles when listening to music, so it's accurate to my listening tendencies, and actual real-time music data.

**Comment by mavproductions** [comment]
Finally got around to installing and testing this! Definitely have a few use-cases :))

I am wondering about the Plex integration with scrobbling. 

Are there limits that would block for "live scrobbling" so when I'm actually listening to the song, on every 5 second update, it also asks last.fm to follow the scrobble status? Plex API or last.fm limited in that regard?

**Reply by mavproductions** [reply]
Right now it just "pushes" the scrobble to Last.fm instead of some sort of currently "scrobbling" method where last.fm friends or other software I have connected to my last.fm (discord friends) can monitor my live listening status.

Having this would be quite a meaningful addition.

**Reply by FoxxMD** [reply]
This feature has [been previously requested](https://github.com/FoxxMD/multi-scrobbler/issues/74) and I have an [open issue](https://github.com/FoxxMD/multi-scrobbler/issues/164) for it. I don't have an ETA for getting it done but is something I've thought about.

**Reply by mavproductions** [reply]
Yahoo! I really do hope you keep at it! This would be huge for me. Love the idea of actively sharing my music activity with others live. Such a great feature to consider. This would blow open some water gates(in a good way!!) :):):)
